### ISSUE TITLE
[Support]: multiple three js bundles. 

### ISSUE THREAD

**Issue by PaleHazy** [Elius94/react-photo-sphere-viewer#39]
### What are you trying to do?

I am trying to figure out why multiple bundles of three js are being included in the build and I think I was able to track it down to a version mismatch in the peer dependencies of https://github.com/Elius94/psv-lens-flare-plugin.

I was wondering if you had some approaches to fix this, other than just keeping all the three versions the same

### Library Version

latest

### What operating system are you using?

Windows

### What browser are you using?

Chrome

**Comment by Elius94** [comment]
Hi! I don't know why it happens. In my Lensflare plugin you can see that three is a peer dependency.

https://github.com/Elius94/psv-lens-flare-plugin/blob/784a7b4ebd78de0f8a5a071d97f2a26ce9ba8207/package.json#L37C5-L37C5

So, I don't really know how to prevent this.
Can you help me?

**Comment by PaleHazy** [comment]
I believe my initial assessment regarding the source of the dependency conflict might have been inaccurate.

One version is located here:

![image](https://github.com/Elius94/react-photo-sphere-viewer/assets/44535172/28de3c64-81a3-4b4f-970c-1fcfd40995df)

And another at the root of `node_modules`:

![image](https://github.com/Elius94/react-photo-sphere-viewer/assets/44535172/b0861b97-dbe1-4cc4-8119-e98be92fa89e)

This duplication results in the inclusion of two distinct `threejs` builds in the final bundle.

Additionally, my application already incorporates a `three` instance, leading to the somewhat amusing situation of having three different `three` bundles

![image](https://github.com/Elius94/react-photo-sphere-viewer/assets/44535172/4598710c-b96e-45ae-812e-920766748e44)

I observed that `react-photo-sphere-viewer` lists `@photo-sphere-viewer` as a direct dependency. When running the build from the repository, I noticed code references to the original package attempting imports from "../node_modules/....".

A potential solution could be to reclassify `@photo-sphere-viewer` and other similar imports as peer dependencies instead of direct dependencies. This approach might immediately resolve the conflict. However, it also poses the inconvenience of requiring users to manually install these dependencies.

Alternatively, could we explore modifying the code to support dynamic imports for all features? This might offer a more streamlined and flexible solution.

![image](https://github.com/Elius94/react-photo-sphere-viewer/assets/44535172/e9d43f58-db4b-44b4-97de-98d708fae46e)

Lastly, it seems that some plugins are imported solely for re-export. In these cases, providing users the option to manually import these plugins from `@photo-sphere-viewer` could be a more efficient approach.
